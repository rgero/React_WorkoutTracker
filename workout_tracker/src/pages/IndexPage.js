import React from "react";

import Container from 'react-bootstrap/Container';

import {DashboardPage} from './DashboardPage';
import {Context as AuthContext} from '../context/AuthContext';

const IndexPage = () => { 
    const {state, tryLocalSignin} = React.useContext(AuthContext);

    React.useEffect(()=> {
        tryLocalSignin();
    }, [])

    if (!state.token) {
        return (
            <Container>
                Welcome to the Workout Tracker. The goal of this app is to help you track your workouts and help you generate some cool stats.
            </Container>
        );
    } else {
        return (<DashboardPage/>);
    }
};

export default IndexPage;