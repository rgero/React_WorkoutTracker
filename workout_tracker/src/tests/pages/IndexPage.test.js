import renderer from 'react-test-renderer';
import IndexPage from '../../pages/IndexPage';

test('Renders Index Page and checks against snapshot', ()=> {
    const indexRender = renderer.create(<IndexPage />).toJSON();
    expect(indexRender).toMatchSnapshot();
})