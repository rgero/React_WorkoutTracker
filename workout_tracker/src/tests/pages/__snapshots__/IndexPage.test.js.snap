// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Index Page and checks against snapshot 1`] = `
<div
  className="container"
>
  Welcome to the Workout Tracker. The goal of this app is to help you track your workouts and help you generate some cool stats.
</div>
`;
